### DISCUSSION TITLE
[item] 显示期刊标签的笔记模板示例

### DISCUSSION THREAD

**Discussion by MuiseDestiny** [windingwind/zotero-better-notes#521] [Note Templates]
以下代码在 Zotero 7.0.0-beta.58+2665090aa、Ethereal Style 4.0.7、Better Notes for Zotero 1.1.4-beta.53上测试可以使用。请确保你的插件更新到最新。

**预览图**
![image](https://github.com/windingwind/zotero-better-notes/assets/51939531/1c8ecbcb-fc4b-4292-9638-14414501c290)

你可以**复制核心代码**作为你其它笔记模板的一部分
```js
${{
let space = " ㅤㅤ ㅤㅤ"
return Array.prototype.map.call(
	Zotero.ZoteroStyle.api.renderCell(topItem, "publicationTags").childNodes,
	e => {
		e.innerText =  space + e.innerText + space;
		return e.outerHTML
	}
	).join(space)
}}$
```

下面是完整BN分享代码，复制后可通过Zotero的`编辑`-`从剪贴板导入笔记模板`导入

```ymal
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Edit->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
name: "[item] Example Publication Tags"
content: |-
  // @use-markdown
  // @author Polygon
  // @link https://github.com/windingwind/zotero-better-notes/discussions/521
  ${{
  let space = " ㅤㅤ ㅤㅤ"
  return Array.prototype.map.call(
  	Zotero.ZoteroStyle.api.renderCell(topItem, "publicationTags").childNodes,
  	e => {
  		e.innerText =  space + e.innerText + space;
  		return e.outerHTML
  	}
  	).join(space)
  }}$
  ```

**Comment by jlu511** [comment]
666666666666

**Comment by yzzmoon** [comment]
作者您好，我想问问为什么我复制过去模板里作者和年份那部分会不成功呢，是哪一部分代码出问题了呢？ @windingwind 不知道您会不会在线多一些，冒昧地艾特您一下，多有打扰，先抱歉啦
![1690272027344](https://github.com/windingwind/zotero-better-notes/assets/140363379/d3bc6d15-0f5c-4cee-8479-8ca9d8c89a9f)

**Comment by evanvong** [comment]
大佬您更新了beta56的适配，模板是否可以更新一下？现在使用模板提示：Template Preview Error: TypeError: Zotero.ZoteroStyle.data.ztoolkit.ItemTree.fieldHooks.globalCache is undefined 。

**Reply by yusaiwen** [reply]
遇到了相同的问题，求帮助

**Reply by jhhj7148** [reply]
看了一下，好像是引用了Ethereal Style插件的标签（Zotero.ZoteroStyle.data.ztoolkit.ItemTree.fieldHooks.globalCache.getFieldHooks.PublicationTags）导致的，可能新版Ethereal Style这个语句有改动

**Reply by yusaiwen** [reply]
我是装了Ethereal Style的一个pre版本所以出了问题，我装回了上一个23年发的版本就正常了～

**Reply by MuiseDestiny** [reply]
已重新编辑笔记模板
